src/users/users.service.ts

```diff
@@ -1,27 +1,20 @@
-import { extname } from 'path';
-import { createHash } from 'crypto';
-import { Role } from '@prisma/client';
-import { Injectable, NotFoundException } from '@nestjs/common';
+import { User } from '@prisma/client';
+import { Injectable } from '@nestjs/common';
 
 import { FindUserDto } from './dto/find-user.dto';
-import { S3Service } from '../service/s3.service';
 import { CreateUserDto } from './dto/create-user.dto';
 import { UpdateUserDto } from './dto/update-user.dto';
 import { PrismaService } from '../prisma/prisma.service';
 import { BcryptService } from '../service/bcrypt.service';
 import { ICreateStrategy } from './create-strategy/icreate.strategy';
 import { UploadProfilePhotoResponse } from './dto/upload-avatar-response.dto';
+import { UsersRepository } from './repository/users.repository';
 
-type UserAttributesFilter = {
-  userAttributes: {
-    [key: string]: { startsWith: string; mode: 'insensitive' };
-  };
-};
 @Injectable()
 export class UsersService {
   constructor(
     private prismaService: PrismaService,
-    private s3Service: S3Service,
+    private usersRepository: UsersRepository,
```

Comment:
@aarbanas Oh, it would be better to import it as a dependency in the `auth.service.ts` then.

Here we rely on a dependency `usersService` which contains the service `bcryptService` which if we want to Unit Test would be a "nightmare" because instead of providing a mock directly we would need to mock two services.